pulsar/internal/connection.go

```diff
@@ -188,21 +188,30 @@ type connection struct {
 	maxMessageSize int32
 }
 
-func newConnection(logicalAddr *url.URL, physicalAddr *url.URL, tlsOptions *TLSOptions,
-	connectionTimeout time.Duration, auth auth.Provider) *connection {
+// ConnectionOptions defines configurations for creating connection.
+type ConnectionOptions struct {
+	LogicalAddr       *url.URL
+	PhysicalAddr      *url.URL
+	TLS               *TLSOptions
+	ConnectionTimeout time.Duration
+	Auth              auth.Provider
+	Logger            log.Logger
+}
```

Comment:
If `ConnectionOptions` is only used as a parameter of `newConnection()`, can we consider it as a private?

Reply:
Sure, will update it